tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/RandomForest.scala

```diff
@@ -80,36 +86,38 @@ class RandomForestTrainer(validationPercentage: Double, numDecisionTrees: Int,
   maximumDepthPerTree: Int = Integer.MAX_VALUE)
     extends ProbabilisticClassifierTrainer {
 
+  private val dtTrainer = new DecisionTreeTrainer(validationPercentage, featuresExaminedPerNode,
+    maximumDepth = maximumDepthPerTree)
+
   /** Induces a RandomForest from a set of feature vectors.
     *
     * @param data a set of feature vectors to use for training
     * @return the induced random forest
     */
   override def apply(data: FeatureVectorSource): ProbabilisticClassifier = {
```

Comment:
I don't quite understand the structure of this function. You create a bunch of DTs, and then you serialize them to disk, and then you read them all back out. So at the end, you have all DTs in memory anyways. What's the problem with having them in memory right from the start then?